### ISSUE TITLE
Error in weightedVar when weight has NA values

### ISSUE THREAD

**Issue by gdemin** [HenrikBengtsson/matrixStats#140]
It seems there is a bug in `weightedVar`. Following example gives me error "if (!all(tmp)) { : missing value where TRUE/FALSE needed":
```R 
vec = c(1, 2, 3, 4, 5)
w = c(1, 1, 1, 1, NA)
matrixStats::weightedVar(vec, w = w, na.rm = TRUE)
```
Expected behavior for `na.rm = TRUE`  is:
```
matrixStats::weightedVar(vec[!is.na(w)], w = w[!is.na(w)], na.rm = TRUE)
```
and NA result when `na.rm = FALSE`.

Session info:
```
R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Linux Mint 19.1

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=ru_RU.UTF-8       LC_NUMERIC=C               LC_TIME=ru_RU.UTF-8        LC_COLLATE=ru_RU.UTF-8    
 [5] LC_MONETARY=ru_RU.UTF-8    LC_MESSAGES=ru_RU.UTF-8    LC_PAPER=ru_RU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=ru_RU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.5.3     tools_3.5.3        packrat_0.5.0      matrixStats_0.54.0
```

**Comment by HenrikBengtsson** [comment]
Thanks, I can reproduce this.  This was missed because missing values are only tested in `x` not weightes `w`.  Will be fixed in next release.

**Comment by HenrikBengtsson** [comment]
Note to self: make sure missing values in weights are handled and documented as in, say,  weightedMeans().

**Comment by HenrikBengtsson** [comment]
This bug applies to:

* [ ] `weightedMad()`
* [ ] `weightedVar()`

The behavior of `weightedMean()` and `weightedMedian()` with missing values in the weights is to return NA.

**Comment by HenrikBengtsson** [comment]
This has been fixed in the develop branch.  Thanks again for reporting on this.

**Comment by HenrikBengtsson** [comment]
This is fixed in matrixStats 0.55.0 which is now on CRAN.
